caffe2/core/tensor.h

```diff
@@ -189,7 +189,6 @@ class CAFFE2_API Tensor final {
    * 'async' parameter triggers async copy for CUDA tensors
    */
   void CopyFrom(const Tensor& src, bool async = false) {
-    AT_ASSERT(!impl_->is_variable());  // TODO: remove this when Variable and Tensor are merged
```

Comment:
like above, this should probably be replaced with a check that `requires_grad=False` instead of just deleted.

Reply:
fixed.